File: easy/go/0053.go
func Max(x, y int)int{
    if x < y{
        return y
    }
    return x
}
func maxSubArray(nums []int) int {
    var dp [20001]int
    var max_ int = nums[0]
    dp[0] = nums[0]
    for i:= 1; i < len(nums); i++{
        dp[i] = Max(dp[i-1] + nums[i], nums[i])
        max_ = Max(dp[i], max_)
    }
    return max_
}

File: easy/cpp/0053.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int size = nums.size();
        int dp[20001] = {0};
        dp[0] = nums[0];
        int max_ = dp[0];
        for(int i = 1 ; i < size; i++){
            dp[i] = nums[i] < dp[i-1] + nums[i] ? dp[i-1] + nums[i] : nums[i];
            max_ = max_ < dp[i] ? dp[i] : max_;
        }
        return max_;
    }
};